databox/client/lib/DataboxApiClient.ts

```diff
@@ -67,24 +66,21 @@ export default class DataboxApiClient {
   /**
    * NOTE: any caller must handle tracking local balances of gift cards and removing them if they're depleted!
    */
-  public async exec<
-    IO extends IFunctionInputOutput,
+  public async query<
     IVersionHash extends keyof ITypes & string = any,
-    IFunctionName extends keyof ITypes[IVersionHash] & string = 'default',
-    ISchemaDbx extends ITypes[IVersionHash][IFunctionName] = IO,
   >(
     versionHash: IVersionHash,
-    functionName: IFunctionName,
-    input: ISchemaDbx['input'],
+    sql: string,
+    boundValues?: any[],
     microPayment: IPayment & {
       onFinalized?(metadata: IDataboxExecResult['metadata'], error?: Error): void;
     } = {},
-  ): Promise<IDataboxExecResult & { output?: ISchemaDbx['output'] }> {
+  ): Promise<IDataboxExecResult & { output?: any[] }> {
```

Comment:
Seems like we should let you pass this in, even if we can't infer it anymore.

Reply:
Agreed

Reply:
Pushed fix.